### PR TITLE
Mock camera improvements

### PR DESCRIPTION
**Pull request by zouppen**
This PR has two related but distinct patches:

* Allow reading mock camera QR code data dynamically from a file or FIFO instead of static content. Makes it easier to debug UI state transformations because app is not needed to restart while testing many QR codes.
* Make mock camera behave more like a real camera by keeping the "opened" state consistent. Without this patch `isOpened()` returns always true, making some operations to fail on mock camera.

### CODE REVIEW

File: lib/mocks/scanner.js [hunk 1]
```diff
@@ -28,16 +31,24 @@ function config (_configuration) {
...
   }, cbTimeout)
 }
 
 function scanMainQR (cryptoCode, callback) {
+  opened = true
   _cancelCb = callback
   handle = setTimeout(function () {
-    var resultStr = mockData.qrData[cryptoCode]
+    opened = false
+    if (mockData.qrDataSource === undefined) {
```
Comment (RafaelTaranto) [inline lib/mocks/scanner.js]: No need to compare it to undefined: 

`if (!mockData.qrDataSource) {`

File: lib/mocks/scanner.js [hunk 2]
```diff
@@ -28,16 +31,24 @@ function config (_configuration) {
...
+      var resultStr = mockData.qrData[cryptoCode]
+    } else {
+      resultStr = fs.readFileSync(path.join(__dirname, "../../", mockData.qrDataSource))
```
Comment (RafaelTaranto) [inline lib/mocks/scanner.js]: Styling change: needs to use single quotes

**Comment by zouppen** [comment]
Thanks for the review, changes applied.

**Comment by joshmh** [comment]
Thanks, @zouppen!

**Pull request merged**